tests/resources/test_predictor.py

```diff
@@ -300,3 +301,39 @@ def test_returned_predictor(valid_graph_predictor_data):
     assert len(result.predictors) == 2
     assert isinstance(result.predictors[0], uuid.UUID)
     assert isinstance(result.predictors[1], ExpressionPredictor)
+
+def test_returned_plain_predictor(valid_simple_ml_predictor_data):
```

Comment:
Ah, yes. I wasn't sure how to communicate this part. Because we mock out the response no matter what, whether you set the pattern to `PLAIN`, `INFER`, or `FORMULATION`, this test and the tests below will always pass. It doesn't matter what kind of predictor is in the mocked response. I tried it out during development. You are absolutely right about all of that.

Now, it feels small, but I think what the team was after here was making sure that `autoconfigure` could recognize all three of these patterns, `PLAIN`, `INFER`, and `FORMULATION`, and _nothing else_. The nothing else is tested above where we tried feeding `autoconfigure` the non-existent `yogurt` pattern:

https://github.com/CitrineInformatics/citrine-python/blob/main/tests/resources/test_predictor.py#L277

`PLAIN` was exercised and accepted in the happy path. We wanted to make sure the other two patterns were accepted with no hassle, as well. Since we got this far, I thought I might as well make the predictor type in the mocked response match the pattern being exercised. But I agree that the tests as written are coming across as confusing. We could:

1. Change the test names to `test_returned_predictor` (reversion), `test_simple_pattern`, `test_formulation_pattern`. Change the docstrings to `"""Exercise auto_configure PLAIN pattern"""` and `"""Exercise auto_configure FORMULATION pattern"""`,  or
2. Revert my updates entirely. 

I prize readability and am fine with either, so let me know what feels best to you as a programmer.

Reply:
If you just want to test that an argument is accepted without error, then the minimum test to do that would be something like:
```
session = FakeSession()
pc = PredictorCollection(uuid.uuid4(), session)
pc.auto_configure(training_data=GemTableDataSource(table_id=uuid.uuid4(), table_version=0), pattern="INFER")
```

But this is another reason to use an enumeration. With an enumeration, that guarantee is baked in.